### PR TITLE
Removing the buffer overflow bug in the IPv6 addresses with prefix le…

### PR DESCRIPTION
**Pull request by metelik**
…ngth of 128

### CODE REVIEW

**Review by alanj853 — APPROVED** [review]

File: src/netif.c [hunk 1]
```diff
@@ -1586,7 +1586,7 @@ static int get_ipaddr6(const struct ip_setting_handler *handler, struct netif *n
                     &ip6.s6_addr[15],
                     &prefix, &scope, &parsed_ifname[0])) {
         char address[INET6_ADDRSTRLEN] = {'\0', };
-        char prefix_str[4]             = {'\0', };
+        char prefix_str[5]             = {'\0', };
```
Comment (PeterJeaiem) [inline src/netif.c]: a define instead of a magic number ?

File: src/netif.c [hunk 2]
```diff
@@ -1602,7 +1602,7 @@ static int get_ipaddr6(const struct ip_setting_handler *handler, struct netif *n
           debug("[%s %d]: Address scope = '%s'", __FILE__, __LINE__, addr_scope);
         }
 
-        sprintf(prefix_str, "/%d", prefix);
+        snprintf(prefix_str, sizeof(prefix_str), "/%d", prefix);
```
Comment (PeterJeaiem) [inline src/netif.c:1606]: can put the magic number replacement  above instead of the sizeof...

File: src/netif.c (hunk 1 above)
Comment (metelik) [reply src/netif.c]: good point

**Review by adrigonzo — APPROVED** [review]

File: src/netif.c (hunk 2 above)
Comment (metelik) [reply src/netif.c:1606]: I prefer sizeof ;) rather than a precompiler macro

Comment (metelik) [reply src/netif.c:1606]: corrected

File: src/netif.c (hunk 1 above)
Comment (metelik) [reply src/netif.c]: corrected

**Pull request merged**